### PR TITLE
update pretty param description

### LINKED ISSUE #120882: Curl request gets pretty printed by default
### What happened?

```
curl http://localhost:8080/api/v1/namespaces/my-namespace/pods/my-pod
```

= 64K bytes

```
curl http://localhost:8080/api/v1/namespaces/my-namespace/pods/my-pod?pretty=false
```

= 32k bytes

### What did you expect to happen?

API call defaults to pretty=false

The docs say: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/
```
pretty	If 'true', then the output is pretty printed.
```

It does not mention that there is a default. So to me, that implies that pretty is not set/true

### How can we reproduce it (as minimally and precisely as possible)?

Curl an object with and without the arg `pretty=true` and diff the response size

### Anything else we need to know?

IF pretty=true is the default, I would then prefer the docs are updated to reflect that. But also reconsider making the x2 response size the default. 

### Kubernetes version

<details>
1.25 EKS
</details>

### Cloud provider

<details>
AWS
</details>

### OS version

_No response_

### Install tools

_No response_

### Container runtime (CRI) and version (if applicable)

_No response_

### Related plugins (CNI, CSI, ...) and versions (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #120882 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go [hunk 1]
```diff
@@ -311,7 +311,7 @@ func (b *builder) buildRoute(root, path, httpMethod, actionVerb, operationVerb s
 		Path(root+path).
 		To(func(req *restful.Request, res *restful.Response) {}).
 		Doc(b.descriptionFor(path, operationVerb)).
-		Param(b.ws.QueryParameter("pretty", "If 'true', then the output is pretty printed.")).
+		Param(b.ws.QueryParameter("pretty", "If 'true', then the output is pretty printed. It's false by default but is true for specific 'human' user agents. This includes curl, wget, etc.")).
```
Comment (Jefftree) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go:314]: Maybe etc is a bit vague? The last one is Mozilla/5.0 which I believe covers all modern web browsers.

**Comment by Jefftree** [comment]
/cc @liggitt 
Wondering if you have any thoughts on this? Thanks!

File: staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go (hunk 1 above)
Comment (carlory) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go]: updated.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go [hunk 2]
```diff
@@ -311,7 +311,7 @@ func (b *builder) buildRoute(root, path, httpMethod, actionVerb, operationVerb s
 		Path(root+path).
 		To(func(req *restful.Request, res *restful.Response) {}).
 		Doc(b.descriptionFor(path, operationVerb)).
-		Param(b.ws.QueryParameter("pretty", "If 'true', then the output is pretty printed.")).
+		Param(b.ws.QueryParameter("pretty", "If 'true', then the output is pretty printed. It's false by default but is true for specific 'human' user agents. This includes curl, wget, and Mozilla/5.0.")).
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go:314]: Would this read more naturally?

> If 'true', then the output is pretty printed. Defaults to 'false' unless the user-agent indicates a browser or command-line HTTP tool (curl and wget).

Comment (carlory) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go]: updated.

**Pull request merged**